src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ITypeInfo.cs

```diff
@@ -8,11 +9,18 @@ namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
     /// <summary>
     /// Contains type metadata.
     /// </summary>
-    public interface ITypeInfo : IMemberInfo
+    public interface ITypeInfo : IMemberInfo, IEquatable<ITypeInfo>
     {
         /// <summary>
         /// Fully qualified name of the type.
         /// </summary>
+        /// <remarks>
+        /// On CoreCLR, we compile against the System.String contract in System.Runtime
```

Comment:
Makes it seem like this FUllName is only for strings. should clarify the string scenario as an example